### PR TITLE
Make kerberos errors more searchable by including error names

### PR DESCRIPTION
**Pull request by richlanc**
https://ldapwiki.com/wiki/Kerberos%20Error%20Codes

### CODE REVIEW

File: src/sspi.rs [hunk 1]
```diff
@@ -1415,11 +1415,78 @@ impl From<Asn1DerError> for Error {
...
+        [0x42] => "KRB_AP_ERR_USER_TO_USER_REQUIRED",
+        [0x43] => "KRB_AP_ERR_NO_TGT",
+        [0x44] => "Unrecognised Domain - KDC_ERR_WRONG_REALM",
+        _ =>  "MISSING_ERROR",
+    }
+}
+
 impl From<KrbError> for Error {
     fn from(err: KrbError) -> Self {
+        let error_code = err.0.error_code.as_unsigned_bytes_be();
```
Comment (SpeeddY) [inline src/sspi.rs:1484]: Trying to find the documentation for this method `as_unsigned_bytes_be`
Please could you link it?

**Review by SpeeddY — APPROVED** [review]
Looks good

**Comment by SpeeddY** [comment]
I have approved this, but it does actually need testing

**Comment by SpeeddY** [comment]
```
[2022-08-11 11:28:45] [DEBUG] (55c3e63671b5) NLA - Starting Authentication Handshake
[2022-08-11 11:28:46] [ERROR] (55c3e63671b5) Error during RDP connection sequence: InternalError: Got the krb error: Unrecognised Username - KDC_ERR_C_PRINCIPAL_UNKNOWN (error code: 100663296)
```

**Comment by SpeeddY** [comment]
googling the error code / string gives the first result of https://docs.microsoft.com/en-us/troubleshoot/windows-server/windows-security/kdc-err-c-principal-unknown-s4u2self-request

**Comment by SpeeddY** [comment]
Happy with this, the new error reporting is an improvement and works well :green_apple:

File: src/sspi.rs (hunk 1 above)
Comment (richlanc) [reply src/sspi.rs:1484]: https://github.com/Devolutions/picky-rs/blob/34e598359de96b8f8301b815efc14e1804e2a728/picky-asn1/src/wrapper.rs#L350

It's "as unsigned bytes big endian"

**Pull request merged**